server/jetstream_cluster.go

```diff
@@ -4336,14 +4339,14 @@ func (s *Server) jsClusteredStreamListRequest(acc *Account, ci *ClientInfo, filt
 		}
 	}
 	// Don't hold lock.
-	js.mu.Unlock()
+	js.mu.RUnlock()
 
 	const timeout = 4 * time.Second
 	notActive := time.NewTimer(timeout)
 	defer notActive.Stop()
 
 LOOP:
-	for {
+	for len(sent) > 0 {
```

Comment:
Well, it looks like we should then do the range streams loop before setting up the internal sub, because if all are offline, then this change here means that we will never wait for responses that we would have gotten in the msg callback.

Reply:
Not following, maybe quick f2f?

Reply:
Got it.